.github/workflows/full_build.yml

```diff
@@ -36,118 +35,244 @@ jobs:
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - uses: actions/download-artifact@v4
+      with:
+        path: ${{runner.workspace}}/acados
+
+    - name: Install Python interface
+      # TODO cache
       working-directory: ${{runner.workspace}}/acados
       shell: bash
       run: |
-        echo "ACADOS_SOURCE_DIR=$(pwd)" >> $GITHUB_ENV
-        echo "ACADOS_INSTALL_DIR=$(pwd)" >> $GITHUB_ENV
-        echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(pwd)/lib" >> $GITHUB_ENV
-        echo "OCTAVE_PATH=$OCTAVE_PATH:$(pwd)/interfaces/acados_matlab_octave:$(pwd)/interfaces/acados_matlab_octave/acados_template_mex" >> $GITHUB_ENV
-        echo "MATLABPATH=$MATLABPATH:$(pwd)/interfaces/acados_matlab_octave:$(pwd)/interfaces/acados_matlab_octave/acados_template_mex" >> $GITHUB_ENV
+        echo "ACADOS_SOURCE_DIR=${{runner.workspace}}/acados" >> $GITHUB_ENV
+        echo "ACADOS_INSTALL_DIR=${{runner.workspace}}/acados" >> $GITHUB_ENV
+        echo "LD_LIBRARY_PATH=${{runner.workspace}}/acados/lib" >> $GITHUB_ENV
+        ${{runner.workspace}}/acados/.github/linux/install_python.sh''
 
-    # test interfaces
-    - name: Install Casadi MATLAB
-      working-directory: ${{runner.workspace}}/acados/external
+    - name: Run CMake python tests (ctest)
+
+      working-directory: ${{runner.workspace}}/acados/build
       shell: bash
-      run: |
-        sudo chmod +x ./../.github/linux/install_casadi_matlab.sh
-        ./../.github/linux/install_casadi_matlab.sh
-        echo "MATLABPATH=$MATLABPATH:$(pwd)/casadi-matlab" >> $GITHUB_ENV
+      # Execute tests defined by the CMake configuration.
+      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+      run: ${{runner.workspace}}/acados/.github/linux/run_tests.sh
 
-    - name: Prepare Octave
+  MATLAB_interface:
+    needs: core_build
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - uses: actions/download-artifact@v4
+      with:
+        path: ${{runner.workspace}}/acados
+
+    - name: Install Casadi MATLAB
       working-directory: ${{runner.workspace}}/acados/external
       shell: bash
       run: |
-        sudo apt-get update
-        sudo apt-get install liboctave-dev -y --fix-missing
-        octave --version
-        ./../.github/linux/install_casadi_octave.sh
-        echo "OCTAVE_PATH=$OCTAVE_PATH:$(pwd)/casadi-octave" >> $GITHUB_ENV
+        sudo chmod +x ${{runner.workspace}}/acados/.github/linux/install_casadi_matlab.sh
+        ${{runner.workspace}}/acados/.github/linux/install_casadi_matlab.sh
+        echo "MATLABPATH=$MATLABPATH:${{runner.workspace}}/acados/interfaces/acados_matlab_octave:${{runner.workspace}}/acados/interfaces/acados_matlab_octave/acados_template_mex:${{runner.workspace}}/acados/external/casadi-matlab" >> $GITHUB_ENV
+        echo "ACADOS_INSTALL_DIR=${{runner.workspace}}/acados" >> $GITHUB_ENV
```

Comment:
The duplication of setting up the environment variables is not so nice.
And the fact that this is part of some other step.
Maybe we could have one script `export_paths.sh` that is executed in each job in one step.
If this sets some paths that are not needed for the particular job, like `MATLABPATH`, it would be fine too.
Just not sure how to get the `{{runner.workspace}}` in there.

Reply:
I just pass a {{runner.workspace}} as an argument to the bash script.